testutil/daemon/daemon.go

```diff
@@ -252,17 +256,24 @@ func (d *Daemon) StartWithLogFile(out *os.File, providedArgs ...string) error {
 		d.pidFile = filepath.Join(d.Folder, "docker.pid")
 	}
 
-	d.args = append(d.GlobalFlags,
-		"--containerd", containerdSocket,
-		"--data-root", d.Root,
+	if d.containerdSocket != "" {
+		d.args = append(d.GlobalFlags,
+			"--containerd", d.containerdSocket,
+			"--data-root", d.Root)
+	} else {
+		d.args = append(d.GlobalFlags,
+			"--data-root", d.Root)
+	}
```

Comment:
Looks like `--data-root` can be added to the list below (as it's unconditionally set), that way we don't need the `else` here, and we can change to

```go
if d.containerdSocket != "" {
	d.args = append(d.GlobalFlags, "--containerd", d.containerdSocket)
}
```

Reply:
The else will go away; sorry about that, I was just messing with something.